src/server/game/AuctionHouseBot/AuctionHouseBotSeller.cpp

```diff
@@ -708,15 +708,15 @@ void AuctionBotSeller::SetPricesOfItem(ItemTemplate const* itemProto, SellerConf
 
     if (sAuctionBotConfig->GetConfig(CONFIG_AHBOT_BUYPRICE_SELLER))
         buyPrice = sellPrice;
+	
+	float basePriceFloat = (buyPrice * stackCount * priceRatio) / (itemProto->GetClass() == 6 ? 200.0f : static_cast<float>(itemProto->GetBuyCount())) / 100.0f;
+    float range = basePriceFloat * 0.04f;
+	
+    buyp = urand(static_cast<uint32>(basePriceFloat - range + 0.5f), static_cast<uint32>(basePriceFloat + range + 0.5f)) + 1;
```

Comment:
Yes, exactly like that.

Looking at the +1 that was originally there (and now 1.5f) looks like it was there to avoid assigning the buy price as 0, which would break things.  I suggest just making it 0.5f, and having a conditional that follows as "if (buyp == 0) then 1 else buyp".  It would make this part of code very clear and explicit on intent.  Any have any alternate opinions?